create-hpc-cluster/mainTemplate.json

```diff
@@ -0,0 +1,559 @@
+      }
+    },
+    "computeNodeNumber": {
+      "type": "string",
+      "defaultValue": "2",
+      "metadata": {
+        "description": "The number of the compute nodes"
+      }
+    },
+    "computeNodeVMSize": {
+      "type": "string",
+      "defaultValue": "Standard_A3",
+      "metadata": {
+        "description": "The VM size for the compute nodes"
+      }
+    },
+    "publicIPName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the public IP address"
+      }
+    },
+    "publicIPRGName": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Resource group for the public IP address resource"
+      }
+    },
+    "publicIPNewOrExisting": {
+      "type": "string",
+      "defaultValue": "new",
+      "allowedValues": [
+        "new",
+        "existing"
+      ],
+      "metadata": {
+        "description": "Indicates whether the public IP address is new or existing"
+      }
+    },
+    "publicIPDNSNameLabel": {
+      "type": "string",
+      "metadata": {
+        "description": "The DNS name label of the public IP address"
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "Administrator user name for the virtual machines and the Active Directory domain."
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Administrator password for the virtual machines and the Active Directory domain"
+      }
+    },
+    "headNodePostConfigScript": {
+      "type": "string",
+      "defaultValue": " ",
+      "metadata": {
+        "description": "Optional URL of a public available PowerShell script you want to run on the head node after it is configured. You can also specify arguments for the script, for example 'http://www.consto.com/mypostscript.ps1 -Arg1 arg1 -Arg2 arg2'."
+      }
+    },
+    "artifactsBaseUrl": {
+      "type": "string",
+      "metadata": {
+        "description": "Base URL for Marketplace",
+        "artifactsBaseUrl": ""
+      }
+    }
+  },
+  "variables": {
+    "apiVersion": "2015-06-15",
+    "computeNodeNumber": "[int(parameters('computeNodeNumber'))]",
+    "cnStorageAccountNumber": "[add(div(sub(variables('computeNodeNumber'), 1), parameters('nbrCNPerStorageAccount')), 1)]",
+    "cnStorageAccountStartIndex": "[add(parameters('storageAccountStartIndex'), 1)]",
+    "storageAccountIndexWidth": "[length(string(sub(add(variables('cnStorageAccountNumber'), variables('cnStorageAccountStartIndex')), 1)))]",
+    "hnStorageAccountName": "[concat(parameters('storageAccountNamePrefix'), padLeft(string(parameters('storageAccountStartIndex')), variables('storageAccountIndexWidth'), '0'))]",
```

Comment:
It is guaranteed in azuredeploy.json. The storageAccountNamePrefix is 16 characters, there is enough name space (a maximum 8 characters for index).